doc/theory/glossary.tex

```diff
@@ -287,6 +287,24 @@
 	
 	The name is inspired from the case when the persistence diagram comes from persistent homology.
 	
+	\subsection*{Heat kernel:} \label{heat kernel}
+	
+	Given a set $\xi$, $k:\xi \times \xi \rightarrow \mathbb{R}$ is said to be a \textit{kernel} if there exists a Hilbert space $H$ with inner product $\langle \cdot, \cdot \rangle$ and a function $\phi: \xi \rightarrow H$, such that $k(x,y) = \langle \phi(x), \phi(y) \rangle$.
+	
+	In contrast to vectorisation methods, kernel methods do not give an explicit feature map. Nevertheless, such methods remain popular for tasks like classification.
+	
+	Let $D, D'$ be two \hyperref[persistence diagram]{persistence diagrams}. Then, the \textit{persistence scale-space kernel}, often called the \textit{heat kernel}, is defined by
+	\begin{equation*}
+	k_\sigma(D,\, D') = \frac{1}{8\pi\sigma}\sum_{p\in D,\, q \in D'}
+	\left(
+	\exp\left(\frac{-\Vert p-q\Vert^2}{8\sigma}\right) -
```

Comment:
Hello, how about this. It is more wordy but maybe ...

\subsection*{Heat kernel:} \label{heat kernel}
	
	Let $(\xi, k)$ be a pair with $\xi$ a set and
	\begin{equation*}
	k:\xi \times \xi \rightarrow \mathbb{R}
	\end{equation*}
	a function.	A \textit{kernel} for this pair is a function
	\begin{equation*}
	\phi: \xi \rightarrow H
	\end{equation*}
	where $\big( H, \langle -, - \rangle \big)$ is an \hyperref[inner product and metric spaces]{inner product space} and $\phi$ satisfies
	\begin{equation*}
	k(x,y) = \big\langle \phi(x), \phi(y) \big\rangle.
	\end{equation*}
	Let $\xi$ be the set of \hyperref[persistence diagram]{persistence diagrams} and $\sigma$ a positive real number. Denoting
	\begin{equation*}
	\Omega = \big\{ (x_1,x_2) \in \mathbb R^2\ |\ x_1 \leq x_2 \big\}
	\end{equation*}
	consider the inner product space of square integrable functions $L^2(\Omega)$ and the map $\phi_\sigma : \xi \to L^2(\Omega)$ described next. Given a persistence diagram $D$, the function $\phi_\sigma(D)$ is the time $\sigma$ solution to the heat equation with initial condition specified by the sum of Dirac delta functions concentrated over the off diagonal points of $D$. More explicitly, the function $\phi_\sigma(D)$ is the time $\sigma$ solution to
	\begin{align*}
	\Delta_x(u) &= \partial_t u && \text{on } \Omega \times \mathbb R_{>0} \\
	u &= 0 && \text{on } \{x_1 = x_2\} \times \mathbb R_{\geq 0} \\
	u &= \sum_{p \in D} \delta_p && \text{on } \Omega \times {0}.
	\end{align*}
	The \textit{heat kernel} is defined by the function $\phi_\sigma$ on the set of persistence diagrams together with the function
	\begin{equation*}
	k_\sigma(D,\, D') = \frac{1}{8\pi\sigma}\sum_{p\in D,\, q \in D'}
	\left(
	\exp\left(\frac{-\Vert p-q\Vert^2}{8\sigma}\right) -
	\exp\left(\frac{-\Vert p-\bar{q}\Vert^2}{8\sigma}\right)
	\right)
	\end{equation*}
	where $\bar{q} = (d,b)$ for a point $q=(b,d)$.
	
	\paragraph{\\ References:} \cite{reininghaus2014stable}

Reply:
I would also suggest expand the entry on "Metric spaces" to "Metric and inner product spaces" and include a hyperreference to "inner product" to this entry in the text above. I wrote a few months ago an entry for L^p spaces that maybe now should be included.

Reply:
I hesitated about introducing the heat equation as well, so i am happy with that. Also, I completely forgot to specify what the set was. Thanks.

The only thing I am not sure about is the definition of kernel. You made me realize that what i wrote in the pr does not correspond to what is actually in the code. I need to have a closer look at both the code and the paper 😞

Reply:
@ammedmar I'm not sure about the proposal. The Main issues I see is that both in functional analysis (https://en.wikipedia.org/wiki/Schwartz_kernel_theorem) and in the theory of reproducing kernel Hilbert spaces (https://en.wikipedia.org/wiki/Reproducing_kernel_Hilbert_space) which is more relevant to the theory of machine learning, "kernel" is always taken to mean some function $k : X \times X \to \mathbb{K}$, where $\mathbb{K}$ is typically $\mathbb{R}$ or $\mathbb{C}$ and $X$ can be just a set with no structure. $k$ should be symmetric ($\mathbb{R}$) or conjugate symmetric ($\mathbb{C}$) and positive definite. If $X$ also has the structure of a vector space, you'd naturally also ask that $k$ be bilinear ($\mathbb{R}$) or sesquilinear ($\mathbb{C}$).

Reply:
@wreise surprised to hear you think there is a mismatch with the code. Could you elaborate? As far as I remember, we really do convolve with Gaussians and then take mirror images. We may disagree with the prefactors in front of the sum and in the exponents, though.

Reply:
@ulupo I am not saying there is. I just confused what the transformed matrix really is, and how i meant to define the matrix in the glossary.

Reply:
@ammedmar in discussing with @wreise, we realized that `HeatKernel` is a wrong name for the transformer we implemented in `gtda.diagrams`. This is because that transformer actually computes the *feature map* on each diagram and not the kernel on pairs of diagrams. Perhaps this led to your proposal as you were trying to connect strongly with the code, so sorry about that!

We are thinking of renaming the `HeatKernel` transformer to just `Heat`. What do you think>

Reply:
I recall us talking about the the word "representation" playing an increasing role. Would it be appropriate to call this something like `heat representation`? In any case, my feel about `Heat` is that it is a bit too naked. I think `heat` should be an adjective instead of a noun.

Reply:
Following your thoughts about the definition of kernel. I think we do not need to define what a general kernel is, just focus on the heat representation map.

Reply:
Thanks @ammedmar. I agree that defining a kernel in generality is not needed given what there is now. We should aim at supporting kernels in future in a similar way as we support different notions of distances in `PairwiseDistance` -- the analogy here being that "kernel is to pairwise distance as vector dot product is to Euclidean distance". This is the content of #356. When that feature is added, we will need to return to this point and address the general notion of a kernel.

As for the naming, I'm hesitant to make it `HeatRepresentation` as: a) longish, b) together with landscapes, silhouettes, betti curves, and persistence images, this "heat thing" is already considered a representation in the latest version of the docs https://giotto-ai.github.io/gtda-docs/dev/modules/diagrams.html#representations, so one might say that "representation" is implicit here. And one might also say that we should then also have `SilhouetteRepresentation` instead of just `Silhouette`, etc.

Reply:
On the naming issue: How about Persistence Image with default option called symmetric and an alternative option called dirichlet?

Reply:
That's a solution, but then we 'remove' the `HeatKernel` transformer (code-wise)? Or modify its core to a real  kernel?
Also, does the default behavior of persistence image correspond to the `symmetric` or `dirichlet`case?

Reply:
About removing the HeatKernel, this proposal moves in that direction, at least the current implementation of the embedding could be absorbed into the P images transformer. The usual P images is the default with parameter `symmetric` and my understanding is that it matches the current implementation. The setting the parameter to `dirichlet` should acts as the current HeatKernel does now. 

Maybe in the future a "Kernel" method will be added, I don't know. If it happens, then one could have different options there too matching the (to be) implemented kernels. Maybe a symmetric_heat and a dirichlet_heat parameter for the kernel transformer will be present. Or there will just be a heat_kernel with the same parameters as the persistence_image transformer (symmetric and dirichlet).

Reply:
@ammedmar thanks for the proposal. Could you please explain the origin of the choice of ``'symmetric'`` to describe the current behaviour of ``PersistenceImage``?

Reply:
Another point is that, if we are to unite everything under the umbrella of `PersistenceImage`, to have less of a documentation and plotting headache all outputs of the new `PersistenceImage` should have the same format. For the current `HeatKernel`, this means in particular passing to the birth-persistence plane and removing the (redundant) information below the diagonal.

Reply:
I can propose a prototype in another PR.

Reply:
About the use of symmetric: Compared with the `HeatKernel`, persistence images remove the boundary by extending the domain from {y >= x} to include {y <= x} with initial condition c(x,y) in the new part given by -c(y,x). Maybe I am just stating something obvious, and missing the core of your question, please reorient. BTW, `antisymmetric` might be even better if not too long for you guys.

Reply:
@ammedmar might be wrong, but I thought the roles of `HeatKernel` and `PersistenceImage` in your message should be reverted. I.e. it is our `HeatKernel` (contrary, btw, to the feature maps in the [paper](https://arxiv.org/abs/1412.6821) it is inspired from, which map to functions on {y >= x}) which plays the antisymmetry trick, while persistence images (both in papers and in our implementation) only convey information about the half-space {y >= x}.

Reply:
I see, I guess I was confused by the paper. What we have then seems to be: an antisymmetrized version (`HeatKernel`) and a rotated version (`PersistenceImage`). I do feel that a single transformer should include both functionalities and that the name `PersistenceImage` is the right name for that transformer. What do you think?